aws/src/main/java/org/apache/iceberg/aws/AwsClientProperties.java

```diff
@@ -66,21 +67,39 @@ public class AwsClientProperties implements Serializable {
    */
   public static final String CLIENT_REGION = "client.region";
 
+  /**
+   * Configure the endpoint to be used to fetch and refresh vended credentials.
+   *
+   * <p>When set, the {@link VendedCredentialsProvider} will be used to fetch and refresh vended
+   * credentials.
+   */
+  public static final String REFRESH_CREDENTIALS_ENDPOINT = "client.refresh-credentials-endpoint";
```

Comment:
shouldn't this be a s3 FileIO property?

Reply:
This property needs to be set before `FileIO` is actually being configured and is similar to the client region. `VendedCredentialsProvider` is configured when `AwsClientProperties#credentialsProvider(..)` is called, so I don't think this property should be a `FileIO` property